central/graphql/resolvers/image_components.go

```diff
@@ -105,24 +155,255 @@ func (resolver *Resolver) ImageComponentCount(ctx context.Context, args RawQuery
+}
+
+func (resolver *imageComponentResolver) Images(ctx context.Context, args PaginatedQuery) ([]*imageResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "Images")
+	return resolver.root.Images(resolver.withImageComponentScope(ctx), args)
+}
+
+func (resolver *imageComponentResolver) ImageVulnerabilityCount(ctx context.Context, args RawQuery) (int32, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "ImageVulnerabilityCount")
+	return resolver.root.ImageVulnerabilityCount(resolver.withImageComponentScope(ctx), args)
+}
+
+func (resolver *imageComponentResolver) ImageVulnerabilityCounter(ctx context.Context, args RawQuery) (*VulnerabilityCounterResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "ImageVulnerabilityCounter")
+	return resolver.root.ImageVulnerabilityCounter(resolver.withImageComponentScope(ctx), args)
+}
+
+func (resolver *imageComponentResolver) ImageVulnerabilities(ctx context.Context, args PaginatedQuery) ([]ImageVulnerabilityResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "ImageVulnerabilities")
+	return resolver.root.ImageVulnerabilities(resolver.withImageComponentScope(ctx), args)
+}
+
+func (resolver *imageComponentResolver) LastScanned(ctx context.Context) (*graphql.Time, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "LastScanned")
+	imageLoader, err := loaders.GetImageLoader(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	q := search.EmptyQuery()
+	q.Pagination = &v1.QueryPagination{
+		Limit:  1,
+		Offset: 0,
+		SortOptions: []*v1.QuerySortOption{
+			{
+				Field:    search.ImageScanTime.String(),
+				Reversed: true,
+			},
+		},
+	}
+
+	images, err := imageLoader.FromQuery(resolver.withImageComponentScope(ctx), q)
+	if err != nil || len(images) == 0 {
+		return nil, err
+	} else if len(images) > 1 {
+		return nil, errors.New("multiple images matched for last scanned image component query")
+	}
+
+	return timestamp(images[0].GetScan().GetScanTime())
+}
+
+// Location returns the location of the component.
+func (resolver *imageComponentResolver) Location(ctx context.Context, args RawQuery) (string, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "Location")
+	var imageID string
+	scope, hasScope := scoped.GetScope(ctx)
+	if hasScope && scope.Level == v1.SearchCategory_IMAGES {
+		imageID = scope.ID
+	} else {
+		var err error
+		imageID, err = getImageIDFromIfImageShaQuery(ctx, resolver.root, args)
+		if err != nil {
+			return "", errors.Wrap(err, "could not determine component location")
+		}
+	}
+
+	if imageID == "" {
+		return "", nil
+	}
+
+	query := search.NewQueryBuilder().AddExactMatches(search.ImageSHA, imageID).AddExactMatches(search.ComponentID, resolver.data.GetId()).ProtoQuery()
```

Comment:
Add a fork on feature flag here. Use the old way, that generates edge ID, if postgres is not enabled, and the this search method if postgres is enabled.